### DISCUSSION TITLE
Can I mangle \ obfuscate import names in the build bundle?

### DISCUSSION THREAD

**Discussion by avalanche1** [parcel-bundler/parcel#5071] [General] [answered]
I'm building a chrome extension with Parcel. According to chrome webstore policy (that hosts Chrome extensions) - Chrome extensions are prohibited to use obfuscation. Unfortunately, for now, we cannot move mission-critical code to a server; and have to keep it inside the extension code, which is very easy to download and read.
Although, Parcel minifies the code - it's still possible to reverse engineer, especially because Parcel doesn't mangle import paths.
For now my bundled files look similar to this (excerpt):
```
        is_js: "YVwU",
        "../../../../../../../utility-belt/helpers/dom/is-el-visible": "NBkX",
        "./preformat-label-text": "uVmy",
        "./find-label-element/find-label-element": "PpOy",
        "../../../../../../../utility-belt/helpers/array/has-contents": "gtdY",
        "../../../../../../../utility-belt/helpers/dom/get-text-content-from-el": "SRGb",
        "./input-is-empty": "xrMH",
        "../../edge-cases/raw-inputs-found": "b9fD",
        "../../edge-cases/associated-labels": "KiPW",
        "../../../../../../../utility-belt/helpers/dom/compare-DOM-positions": "lAHL",
        "../../../../../../data/dictionary/BANKS": "CVde",
        "../../../../../../../utility-belt/helpers/dom/DOM-shortcuts": "lN5N",
```
As you can see - from file names and paths, with some effort it can be deduced what each module does, and also it's hash name. From hash name its contents are found and read.
It would really help if those imports be obfuscated or encoded. Is there a way to achieve that?

**Comment by mischnic** [comment, marked as answer]
> Is there a way to achieve that?

Using scope-hoisting.

**Reply by avalanche1** [reply]
I use scope hoisting extensively. With functions.
Don't see how it applies to modules, though.

**Reply by mischnic** [reply]
https://parceljs.org/cli.html#enable-experimental-scope-hoisting/tree-shaking-support

https://v2.parceljs.org/features/scope-hoisting/

**Reply by avalanche1** [reply]
That helps and not.
It really achieves what I need in terms of getting rid of explicit import names; but I cannot use it because of the ['buffer bug'](https://github.com/formium/formik/issues/2445).
Guess, my only option is to wait for arrival of stable Parcel v2.
